[7.1] Disconnect if `configure_connection` failed (#54710)
rails/rails: activerecord/CHANGELOG.md

*   Fix error handling during connection configuration.

    Active Record wasn't properly handling errors during the connection configuration phase.

>     This could lead to partially configured connection being used, resulting is various exception,

Reviewer: ```suggestion
    This could lead to a partially configured connection being used, resulting in various exceptions,
```

>     the most common being with the PostgresAdapter raising `undefined method `key?' for nil`

Reviewer: ```suggestion
    the most common being with the PostgreSQLAdapter raising `undefined method `key?' for nil`
```

>     of `TypeError: wrong argument type nil (expected PG::TypeMap)`.

Reviewer: ```suggestion
    or `TypeError: wrong argument type nil (expected PG::TypeMap)`.
```

I think this is what you meant here?
